scripts/deploy/index.ts

```diff
@@ -39,18 +40,38 @@ export async function deploy(rpc: JsonRpc, dependencies: Dependencies) {
 	return { uniswapFactory, dai, vat, pot, daiJoin, ethJoin, daiHrd, setDsr }
 }
 
-async function deployToLocalhost() {
+export async function deployAndCache(rpc: JsonRpc, dependencies: Dependencies) {
+	const deployments = await deploy(rpc, dependencies)
+	await ensureDirectoryExists(testDbDirectory)
+	const cacheContents = Object.entries(deployments).reduce((addresses, [key, value]) => ({...addresses, [key]: value.address }), {})
+	await fs.writeFile(testAddressesPath, JSON.stringify(cacheContents, (_, value) => (typeof value === 'bigint') ? `0x${value.toString(16)}`: value))
+	return deployments
+}
+
+export async function deployToLocalhost(signer: MnemonicSigner | PrivateKeySigner) {
 	const jsonRpcEndpoint = 'http://localhost:8545'
 	const gasPrice = 10n**9n
-	const mnemonic = ['zoo', 'zoo', 'zoo', 'zoo', 'zoo', 'zoo', 'zoo', 'zoo', 'zoo', 'zoo', 'zoo', 'wrong']
-	const signer = await MnemonicSigner.create(mnemonic)
 	const rpc = new FetchJsonRpc(jsonRpcEndpoint, fetch, async () => gasPrice, async () => signer.address, signer.sign)
 	const dependencies = new DependenciesImpl(rpc)
-	await deploy(rpc, dependencies)
+	await deployAndCache(rpc, dependencies)
+}
+
+export async function deployToLocalhostWithPrivateKeySigner() {
+	const signer = await PrivateKeySigner.create(0xfae42052f82bed612a724fec3632f325f377120592c75bb78adfcceae6470c5an)
```

Comment:
Seems burdensome to switch private keys up based on environment, let's just standardize the instantseal to use the same accounts:

https://github.com/Keydonix/parity-instantseal/pulls